pkg/client/fallback/fallback.go

```diff
@@ -0,0 +1,51 @@
+package fallback
```

Comment:
What's the reason for the specific `fallback` implementation. What does the `selfhosted` client provide as a first option that the `oci` client couldn't?

Looking at it, I think the only thing would be manifest schema 1 which ggcr doesn't support? We could probably roll that into the `oci` client. Then we'd have manifest schema 1 support + all the nice trappings of ggcr, like automatically detecting credentials in the environment and better auth handling.

Reply:
Two reasons really - I didn't want to break existing users, just fix things for where it currently fails, and the selfhosted client gets metadata back in a single call.